clouddrift/wavelet.py

```diff
@@ -15,22 +15,52 @@
 not the MATLAB implementation is licensed under CloudDrift's MIT license.
 """
 
+import typing as t
+
 import numpy as np
 from scipy.special import gamma as _gamma
 from scipy.special import gammaln as _lgamma
 
 
+@t.overload
```

Comment:
What is this for? @t.overload then a repeat of the definition?

Reply:
These likely fit better in a stub file but the purpose of it is two fold.
1. Document that the output type is directly dependent on the the value of the `complex` parameter
2. Helps the type checker understand the functions output type is dependent on the `complex` parameters value.

Without this, the type checker will think function calls made to `morse_wavelet_transform` can return either type (tuple or a single ndarray) and so will warn us to write code to check the types and handle the two scenarios separately.